homeassistant/components/ws980wifi/sensor.py

```diff
@@ -0,0 +1,322 @@
+    PRESSURE_HPA,
+    SPEED_METERS_PER_SECOND,
+    TEMP_CELSIUS,
+    UNIT_PERCENTAGE,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BUFFER_SIZE: str = "buffer_size"
+LENGTH_MILLIMETERS: str = "mm"
+ILLUMINANCE: str = "lux"
+UV_VALUE: str = "uW/m²"
+UV_INDEX: str = "UV Index"
+DEGREE: str = "°"
+
+DEFAULT_BUFFER_SIZE = 1024
+DEFAULT_NAME = "WS980WiFi"
+DEFAULT_TIMEOUT = 10
+DEFAULT_PORT = 45000
+DEFAULT_SCAN_INTERVAL = 30
+
+ATTRIBUTION = "ELV WiFi-Wetterstation WS980WiFi"
+
+SENSOR_PROPERTIES = {
+    "inside_temperature": [
+        "inside temperature",
+        TEMP_CELSIUS,
+        DEVICE_CLASS_TEMPERATURE,
+        "7",
+        "2",
+        "10",
+    ],
+    "outside_temperature": [
+        "outside temperature",
+        TEMP_CELSIUS,
+        DEVICE_CLASS_TEMPERATURE,
+        "10",
+        "2",
+        "10",
+    ],
+    "dew_point": ["dew point", TEMP_CELSIUS, DEVICE_CLASS_TEMPERATURE, "13", "2", "10"],
+    "apparent_temperature": [
+        "apparent temperature",
+        TEMP_CELSIUS,
+        DEVICE_CLASS_TEMPERATURE,
+        "16",
+        "2",
+        "10",
+    ],
+    "heat_index": [
+        "heat index",
+        TEMP_CELSIUS,
+        DEVICE_CLASS_TEMPERATURE,
+        "19",
+        "2",
+        "10",
+    ],
+    "inside_humidity": [
+        "inside humidity",
+        UNIT_PERCENTAGE,
+        DEVICE_CLASS_HUMIDITY,
+        "22",
+        "1",
+        "1",
+    ],
+    "outside_humidity": [
+        "outside humidity",
+        UNIT_PERCENTAGE,
+        DEVICE_CLASS_HUMIDITY,
+        "24",
+        "2",
+        "1",
+    ],
+    "pressure_absolute": [
+        "pressure absolute",
+        PRESSURE_HPA,
+        DEVICE_CLASS_PRESSURE,
+        "26",
+        "2",
+        "10",
+    ],
+    "pressure_relative": [
+        "pressure relative",
+        PRESSURE_HPA,
+        DEVICE_CLASS_PRESSURE,
+        "29",
+        "2",
+        "10",
+    ],
+    "wind_direction": ["wind direction", DEGREE, None, "32", "2", "1"],
+    "wind_speed": ["wind speed", SPEED_METERS_PER_SECOND, None, "35", "2", "10"],
+    "gust": ["gust", SPEED_METERS_PER_SECOND, None, "38", "2", "10"],
+    "rain": ["rain", LENGTH_MILLIMETERS, None, "41", "4", "10"],
+    "rain_day": ["rain day", LENGTH_MILLIMETERS, None, "46", "4", "10"],
+    "rain_week": ["rain week", LENGTH_MILLIMETERS, None, "51", "4", "10"],
+    "rain_month": ["rain month", LENGTH_MILLIMETERS, None, "56", "4", "10"],
+    "rain_year": ["rain year", LENGTH_MILLIMETERS, None, "61", "4", "10"],
+    "rain_total": ["rain total", LENGTH_MILLIMETERS, None, "66", "4", "10"],
+    "light": ["light", ILLUMINANCE, None, "71", "4", "10"],
+    "uv_value": ["uv value", UV_VALUE, None, "76", "2", "10"],
+    "uv_index": ["uv index", UV_INDEX, None, "79", "1", "1"],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler